include/mxnet/c_api.h

```diff
@@ -373,6 +379,25 @@ MXNET_DLL int MXNDArrayLoad(const char* fname,
                             NDArrayHandle** out_arr,
                             mx_uint *out_name_size,
                             const char*** out_names);
+/*!
+ * \brief Load list of narray from memory.
+ * This will load a list of ndarrays in a similar
+ * manner to MXNDArrayLoad, however, it loads from
+ * pointer in memory, rather than a file.
+ * \param buf pointer to the start of the ndarrays
+ * \param size size of the buffer
+ * \param out_size number of narray loaded.
+ * \param out_arr head of the returning narray handles.
+ * \param out_name_size size of output name arrray.
+ * \param out_names the names of returning NDArrays, can be NULL
+ * \return 0 when success, -1 when failure happens
+ */
+MXNET_DLL int MXNDArrayLoadListFromMemory(const void *buf,
```

Comment:
why do we need this?

Reply:
Without it it becomes difficult to put all your resources in 1 file. For instance we are a text to speech company and we need to keep all the resources for a voice in one file. One could also resolve the problem by using a file pointer instead of a buffer,  but I thought this would be a more general approach.

Reply:
As it would allow you to use encrypted or compressed formats too

Reply:
We already have an API for loading and saving list/dict of NDArrays

Reply:
In the c_api.h the only function that looked right for working from memory was MXNDArrayLoadFromRawBytes but that seems to only work with 1 array rather than loading a list/dict unless I misunderstood?

Reply:
I was looking over the predict API and there are functions there that can get this data (in a few steps), but should it not be in the main api?